data/account/participationRegistry.go

```diff
@@ -806,6 +825,17 @@ func scanRecords(rows *sql.Rows) ([]ParticipationRecord, error) {
 			}
 		}
 
+		if len(rawStateProof) > 0 {
+			stateProof := merklekeystore.Signer{}
+			err = protocol.Decode(rawStateProof, &stateProof)
+			if err != nil {
+				return nil, fmt.Errorf("unable to decode stateproof: %w", err)
+			}
+			var stateProofPk StateProofPublicKey
+			copy(stateProofPk[:], stateProof.GetVerifier()[:])
```

Comment:
So when you `Insert` the participation into the registry you store the `SignerContext` in its entirety, but in the cache only the verifier is stored, and when retrieved from the DB only the verifier is extracted. Doesn't that seem a bit inconsistent and confusing?
The only place the entire `SignerContext` is used it when calling `GetForRound()`.
I think maybe we should completely separate the StateProof data from the Participation data, even store it in a different table `(ParticipationID, StateProofData)`. But we can talk about it for a different PR I guess.